tests/test_schema.py

```diff
@@ -382,6 +405,68 @@ def test_bad_user_defined(self) -> None:
         with pytest.raises(TypeError):
             adapter.validate_python(test)
 
+    def test_valid_nested_user_defined(self) -> None:
+        test = copy.deepcopy(self.base)
+        # Allow any arbitrary JSON as long as all the leaves are valid
+        # FloatFunctionTable
+        test["Parameterisation"]["User-defined"] = {
+            "My model 1": {
```

Comment:
perhaps also a test for the case where the nested headings have to be strings?
```python
 test["Parameterisation"]["User-defined"] = {
            4: {
                   "Function 1": "2.0 * x",
            },
}
```
should raise an error